code/parse/sexp_container.cpp

```diff
@@ -1425,3 +1425,89 @@ int sexp_container_eval_change_sexp(int op_num, int node)
 		return SEXP_FALSE;
 	}
 }
+
+int collect_container_values(int node,
+	SCP_vector<std::pair<const char*, int>> &argument_vector,
+	bool just_count,
+	const char *sexp_name,
+	bool map_keys_only)
+{
+	Assertion(node != -1, "%s wasn't given any ontainers. Please report!", sexp_name);
+
+	// filter out duplicates, but keys/data are case-sensitive
+	// unordered is also ok, because the for-* SEXPs are "any-of"
+	SCP_unordered_set<SCP_string> arguments;
```

Comment:
Why is this a set?
As a user, I'd assume that a duplicate entry in a container will also be evaluated twice. It might lead to unintuitive behaviour if it is silently filtered.
While I personally would like both in-order evaluation and duplicate-including evaluation, there is a case to be made that a user usually doesn't wants this kind of duplicate evaluation (though I could think of a few cases where it would be nice to have). But if you end up wanting to filter the duplicates here, it needs to be clearly documented in the SEXP help text (and while you're on it, lso specifying that it does not adhere to the order of the data in the container would be nice), or some user will scratch their head later on trying to figure this out.

Reply:
Hmm, I think I'd misinterpreted the `already-dupped` function argument in `test_argument_vector_for_condition()` to mean it was doing de-duplication of the argument list, but I found out later it was referring to memory management.

I'll get rid of this.

As a bonus, it'll simplify the code some.